amazon-kinesis-client/src/main/java/software/amazon/kinesis/lifecycle/ShutdownTask.java

```diff
@@ -119,6 +124,7 @@ public TaskResult call() {
                     if (!CollectionUtils.isNullOrEmpty(childShards)) {
                         createLeasesForChildShardsIfNotExist();
                         updateLeasesForChildShards();
+                        garbageCollectLeaseForParentShard();
```

Comment:
1. Added check to use value persisted in lease
2. To differentiate the edge case, added a new state in ShutdownTask instead of sentinel value, can use the sentinel if you feel strongly about it.